Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.h

```diff
@@ -255,28 +381,274 @@ class Wiimote : public ControllerEmu::EmulatedController
   void UpdateButtonsStatus();
 
   void GetButtonData(u8* data);
-  void GetAccelData(u8* data, const ReportFeatures& rptf);
-  void GetIRData(u8* data, bool use_accel);
-  void GetExtData(u8* data);
-
-  bool HaveExtension() const;
-  bool WantExtension() const;
+  void GetAccelData(u8* data);
+  void UpdateIRData(bool use_accel);
+  void UpdateExtData();
 
 private:
-  struct ReadRequest
+  I2CBus m_i2c_bus;
+
+  ExtensionPort m_extension_port{m_i2c_bus};
+
+  struct IRCameraLogic : public I2CSlave
   {
-    // u16 channel;
-    u32 address, size, position;
-    u8* data;
-  };
+    struct
+    {
+      // Contains sensitivity and other unknown data
+      u8 data[0x33];
+      u8 mode;
+      u8 unk[3];
+      // addr 0x37
+      u8 camera_data[36];
+      u8 unk2[165];
+    } reg_data;
+
+    static_assert(0x100 == sizeof(reg_data));
+
+    static const u8 DEVICE_ADDR = 0x58;
+
+    int BusRead(u8 slave_addr, u8 addr, int count, u8* data_out) override
+    {
+      if (DEVICE_ADDR != slave_addr)
+        return 0;
+
+      return RawRead(&reg_data, addr, count, data_out);
+    }
+
+    int BusWrite(u8 slave_addr, u8 addr, int count, const u8* data_in) override
+    {
+      if (DEVICE_ADDR != slave_addr)
+        return 0;
+
+      return RawWrite(&reg_data, addr, count, data_in);
+    }
+
+  } m_camera_logic;
+
+  class ExtensionLogic : public ExtensionAttachment
```

Comment:
Are those magic numbers in here important enough to get either a comment or a named constant?